### PR TITLE
Add two formatters: text, teamcity.

### PR DESCRIPTION
**Pull request by memelet**
Closes https://github.com/stoplightio/spectral/issues/823 and https://github.com/stoplightio/spectral/issues/822.

The text formatter is close to the stylish but with no special formatting and a source:line:column suitable for navigation within an IDE console (IntelliJ specifically).

The teamcity formatter emits teamcity inspection service messages that teamcity will automatically detect and add to the build inspection tab.

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/formatters/teamcity.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { Dictionary } from '@stoplight/types';
+import { IRuleResult } from '../types';
+import { Formatter } from './types';
+import { getSeverityName, groupBySource, sortResults } from './utils';
+
+function escapeString(str?: string | number) {
+  if (!str) {
```
Comment (P0lip) [inline src/formatters/teamcity.ts]: What about `0`?
Let's adjust the condition a bit, suggest the following
```suggestion
  if (str === void 0) {
    return '';
  }
```

File: src/formatters/teamcity.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { Dictionary } from '@stoplight/types';
```
Comment (P0lip) [inline src/formatters/teamcity.ts]:
```suggestion
import { Dictionary, Optional } from '@stoplight/types';
```

File: src/formatters/teamcity.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { IRuleResult } from '../types';
+import { Formatter } from './types';
+import { getSeverityName, groupBySource, sortResults } from './utils';
+
+function escapeString(str?: string | number) {
```
Comment (P0lip) [inline src/formatters/teamcity.ts]:
```suggestion
function escapeString(str: Optional<string | number>) {
```
Let's make the argument required.

File: src/formatters/teamcity.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if (!str) {
+    return '';
+  }
+  return str
```
Comment (P0lip) [inline src/formatters/teamcity.ts]: Being a bit paranoid :sweat_smile: 
```suggestion
  return String(str)
```

File: src/formatters/teamcity.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    .toString()
```
Comment (P0lip) [inline src/formatters/teamcity.ts]:
```suggestion
```

File: .gitignore [hunk 6]
```diff
@@ -26,6 +26,7 @@
 .env.production.local
 .vscode
 .idea
+.nvmrc
```
Comment (P0lip) [inline .gitignore]: Actually, go ahead and commit it.
I'm a happy user of nvm as well :slightly_smiling_face:.

Comment (memelet) [reply .gitignore]: fyi: version is 10.15.3, and all tests pass with that

**Pull request merged**